Description: Pull patch from upstream to fix eval order when byte compiled.
Author: Barry deFreese <bdefreese@debian.org>
Index: haxe-3.0~svn6707/interp.ml
===================================================================
--- haxe-3.0~svn6707.orig/interp.ml	2013-07-23 12:59:39.776241289 -0400
+++ haxe-3.0~svn6707/interp.ml	2013-07-23 12:59:39.772241269 -0400
@@ -484,7 +484,10 @@
 	let vm = Extc.dlcall1 (load "neko_vm_alloc") null in
 	ignore(Extc.dlcall1 (load "neko_vm_select") vm);
 	let loader = Extc.dlcall2 (load "neko_default_loader") null null in
-	let loadprim = Extc.dlcall2 (load "neko_val_field") loader (Extc.dlcall1 (load "neko_val_id") (Extc.dlstring "loadprim")) in
+	let loadprim =
+	let l1 = load "neko_val_field" in
+	let l2 = Extc.dlcall1 (load "neko_val_id") (Extc.dlstring "loadprim") in
+	Extc.dlcall2 (l1) loader (l2) in 
 
 	let callN = load "neko_val_callN" in
 	let callEx = load "neko_val_callEx" in
@@ -4542,4 +4545,4 @@
 decode_expr_ref := decode_expr;
 encode_clref_ref := encode_clref;
 enc_string_ref := enc_string;
-enc_hash_ref := enc_hash
\ No newline at end of file
+enc_hash_ref := enc_hash
Index: haxe-3.0~svn6707/matcher.ml
===================================================================
--- haxe-3.0~svn6707.orig/matcher.ml	2013-07-23 12:59:39.776241289 -0400
+++ haxe-3.0~svn6707/matcher.ml	2013-07-23 12:59:39.772241269 -0400
@@ -534,7 +534,8 @@
 		pc_sub_vars = None;
 		pc_reify = false;
 	} in
-	loop pctx e t, pctx.pc_locals
+	let x = loop pctx e t in
+	x, pctx.pc_locals 
 
 let get_pattern_locals ctx e t =
 	try
@@ -1304,4 +1305,4 @@
 	end;
 ;;
 match_expr_ref := match_expr;
-get_pattern_locals_ref := get_pattern_locals
\ No newline at end of file
+get_pattern_locals_ref := get_pattern_locals
Index: haxe-3.0~svn6707/libs/ttflib/Makefile
===================================================================
--- haxe-3.0~svn6707.orig/libs/ttflib/Makefile	2013-07-18 12:39:59.000000000 -0400
+++ haxe-3.0~svn6707/libs/ttflib/Makefile	2013-07-23 13:01:50.788890945 -0400
@@ -4,11 +4,16 @@
 
 OUTPUT=ttf
 
-all:
+all: native bytecode
+
+native:
 	ocamlopt $(FLAGS) $(FILES:=.ml) -g -a -o ttf.cmxa
 
+bytecode:
+	ocamlc $(FLAGS) $(FILES:=.ml) -g -a -o ttf.cma
+
 exec:
 	ocamlopt $(FLAGS) $(LIBS:=.cmxa) $(FILES:=.ml) main.ml -g -o $(OUTPUT)
 
 clean:
-	rm -rf ttf.cmxa ttf.lib ttf.a $(wildcard *.cmx) $(wildcard *.obj) $(wildcard *.o) $(wildcard *.cmi)
\ No newline at end of file
+	rm -rf ttf.cmxa ttf.lib ttf.a $(wildcard *.cmx) $(wildcard *.obj) $(wildcard *.o) $(wildcard *.cmi)
